File: CPP/ByTier/D/1821_D_Black_Cells.cpp
// Problem: CF 1821 D - Black Cells
// https://codeforces.com/contest/1821/problem/D

/*
Algorithm: Greedy + Two Pointers Approach
Time Complexity: O(n), where n is the number of segments
Space Complexity: O(1), only using a few variables for computation

The problem involves coloring cells using a pointer and a "Shift" button.
We are given segments where we can color cells.
We want to find the minimum number of moves to color at least k cells.

Approach:
1. For each segment:
   - If segment has length 1, increment counter c (single cell segments).
   - Otherwise, add segment length to sum and increment counter c2 (multi-cell segments).
2. For each segment, check if total colored cells (sum + c) is sufficient to meet k:
   - If sum >= k, compute minimum moves to color exactly k cells.
   - If sum + c >= k, compute moves to color remaining cells.
3. Return the minimum moves found, or -1 if impossible.
*/

#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;
#define ll long long
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  int t;
  cin >> t;
  while (t--) {
    int n, k;
    cin >> n >> k;
    vector<int> l(n), r(n);
    for (int i = 0; i < n; ++i)
      cin >> l[i];
    for (int i = 0; i < n; ++i)
      cin >> r[i];
    int sum = 0;
    int ans = 2e9;
    int c = 0;
    int c2 = 0;
    for (int i = 0; i < n; ++i) {
      int len = r[i] - l[i] + 1;
      if (len == 1) {
        c++;
      } else {
        sum += len;
        c2++;
      }
      if (sum >= k) {
        // If we can color k cells using only multi-cell segments
        ans = min(ans, r[i] - (sum - k) + 2 * c2);
      } else if (sum + c >= k) {
        // If we need to use single-cell segments to complete coloring
        ans = min(ans, r[i] + 2 * c2 + 2 * (k - sum));
      }
    }
    cout << (ans == 2e9 ? -1 : ans) << "\n";
  }
  return 0;
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/D/1821_D_Black_Cells.py
# Problem: CF 1821 D - Black Cells
# https://codeforces.com/contest/1821/problem/D

"""
Algorithm: Greedy + Two Pointers
Time Complexity: O(n)
Space Complexity: O(1)

The problem involves coloring cells in segments using a pointer and a shift button.
We want to find the minimum number of moves to color at least k cells.

Key insight:
- To minimize moves, we should try to maximize the number of cells we can color in one shift.
- If we can cover k cells using exactly 'i' segments, we can compute the minimal moves.
- The number of moves is composed of:
  - Moving right to reach a segment: 2 * (i + 1) for i segments
  - Using shift to color: we subtract extra cells we don't need (x = c - k)
  - Minimizing over possible segment combinations.

The algorithm proceeds by checking prefix of segments and computing minimal moves needed.
"""

import sys

input = sys.stdin.readline


def inp():
    return int(input())


def inlt():
    return list(map(int, input().split()))


def insr():
    s = input()
    return list(s[: len(s) - 1])


def invr():
    return map(int, input().split())


for _ in range(inp()):
    n, k = invr()
    l = inlt()
    r = inlt()
    total = 0
    for i in range(n):
        total += r[i] - l[i] + 1
    if k > total:
        print(-1)
        continue
    if k == total:
        print(r[-1] + n * 2)
        continue
    ans = float("inf")
    c, p = 0, 0
    for i in range(n):
        p += l[i] == r[i]  # Count segments with single cell
        c += r[i] - l[i] + 1  # Cumulative count of cells
        if c < k:
            continue
        x = c - k  # Extra cells beyond what we need
        # Compute moves needed for current prefix of segments
        ans = min(ans, 2 * (i + 1) + r[i] - x - min(x, p))
    print(ans)


# https://github.com/VaHiX/CodeForces/